titus-client/src/main/java/com/netflix/titus/runtime/connector/jobmanager/JobEventPropagationUtil.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.runtime.connector.jobmanager;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.netflix.titus.api.jobmanager.model.job.Job;
+import com.netflix.titus.api.jobmanager.model.job.JobDescriptor.JobDescriptorExt;
+import com.netflix.titus.api.jobmanager.model.job.Task;
+import com.netflix.titus.common.util.event.EventPropagationTrace;
+import com.netflix.titus.common.util.event.EventPropagationUtil;
+import com.netflix.titus.common.util.time.Clock;
+
+/**
+ * Help utils for measuring job event notification propagation delays.
+ */
+public class JobEventPropagationUtil {
+
+    public static final String CHECKPOINT_TJC_INTERNAL = "tjcInternal";
+    public static final String CHECKPOINT_GATEWAY_CLIENT = "gwClient";
+    public static final String CHECKPOINT_GATEWAY_INTERNAL = "gwInternal";
+    public static final String CHECKPOINT_FED_CLIENT = "fedClient";
+    public static final String CHECKPOINT_FED_INTERNAL = "federationInternal";
```

Comment:
I don't see this being used at all. Is this only intended for cases where more was being done in the Federation?

Reply:
It is used in `JobEventPropagationMetrics` to create a tag value. It is passed via `FEDERATION_LABELS`